shell/server/proxy.js

```diff
@@ -1649,24 +1643,84 @@ Proxy.prototype.handleRequest = function (request, data, response, retryCount) {
     var path = request.url.slice(1);  // remove leading '/'
     var session = self.getSession(request);
 
-    if (request.method === "GET") {
+    function optionsForBody() {
+      return {
+        content: data,
+        encoding: request.headers["content-encoding"],
+        mimeType: request.headers["content-type"]
+      };
+    }
+
+    if (request.method === "GET" || request.method === "HEAD") {
```

Comment:
Does Node actually ever pass in "HEAD" requests? I was under the impression that it would generate a "GET" request and then chop off the body itself.